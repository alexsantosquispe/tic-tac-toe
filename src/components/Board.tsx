import { useCallback } from 'react';
import useGame from '../hooks/useGame';
import Square from './Square';

export const Board = () => {
  const { data, currentPlayer, checkSquare, combHighLight, winner, isDraw } =
    useGame();

  const handleClick = useCallback(
    (index: number) => {
      checkSquare(index);
    },
    [checkSquare]
  );

  return (
    <div className="flex h-80 w-80 rounded-2xl border border-neutral-200 bg-neutral-100 p-3 shadow-lg md:h-96 md:w-96">
      <div
        className="flex flex-1 flex-wrap overflow-hidden rounded-xl border border-dashed border-neutral-200 bg-white"
        aria-label="Tic Tac Toe board"
      >
        {data.map((square, index) => {
          return (
            <Square
              key={index}
              value={square}
              onClickItem={() => handleClick(index)}
              highLight={combHighLight.includes(index)}
              isDisabled={!!winner || isDraw}
              currentPlayer={currentPlayer}
            />
          );
        })}
      </div>
    </div>
  );
};
