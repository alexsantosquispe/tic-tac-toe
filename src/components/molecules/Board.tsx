import Square from './Square';
import useGame from '../../hooks/useGame';

export const Board = () => {
  const { data, currentPlayer, checkSquare, combHighLight, winner, isDraw } =
    useGame();

  return (
    <div className="flex h-[20rem] w-[20rem] rounded-2xl border border-neutral-200 bg-neutral-100 p-3 shadow-lg md:h-[28rem] md:w-[28rem]">
      <div
        className="flex flex-1 flex-wrap overflow-hidden rounded-xl border border-dashed border-neutral-200 bg-white"
        aria-label="Tic Tac Toe board"
      >
        {data.map((square, index) => {
          return (
            <Square
              key={index}
              index={index}
              value={square}
              onClickItem={checkSquare}
              highLight={combHighLight.includes(index)}
              isDisabled={!!winner || isDraw}
              currentPlayer={currentPlayer}
            />
          );
        })}
      </div>
    </div>
  );
};
