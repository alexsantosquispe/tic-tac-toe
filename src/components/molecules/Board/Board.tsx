import useGame from '../../../hooks/useGame';
import Square from '../Square/Square';

export const Board = () => {
  const { data, currentPlayer, checkSquare, combHighLight, winner, isDraw } =
    useGame();

  return (
    <div
      data-testid="board"
      className="flex h-[20rem] w-[20rem] rounded-2xl border border-neutral-200 bg-neutral-100 p-3 shadow-lg md:h-[28rem] md:w-[28rem] dark:border-white/20 dark:bg-white/10"
    >
      <div
        className="flex flex-1 flex-wrap overflow-hidden rounded-xl border border-dashed border-neutral-200 bg-white dark:border-white/20 dark:bg-black"
        aria-label="Tic Tac Toe board"
      >
        {data.map((square, index) => {
          return (
            <Square
              key={index}
              index={index}
              value={square}
              onClickItem={checkSquare}
              highLight={combHighLight.includes(index)}
              isDisabled={!!winner || isDraw}
              currentPlayer={currentPlayer}
            />
          );
        })}
      </div>
    </div>
  );
};
