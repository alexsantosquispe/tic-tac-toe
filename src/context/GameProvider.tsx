import { useState, type ReactNode } from 'react';
import { DEFAULT_DATA } from '../constants';
import type { SquareValueTypes } from '../types';
import { getWinnerResult } from '../utils';
import GameContext from './GameContext';

interface GameProviderProps {
  children: ReactNode;
}

const GameProvider = ({ children }: GameProviderProps) => {
  const [currentPlayer, setCurrentPlayer] = useState<'X' | 'O'>('X');
  const [data, setData] = useState<SquareValueTypes[]>(DEFAULT_DATA);
  const [winner, setWinner] = useState<SquareValueTypes | null>(null);
  const [combHighLight, setCombHighLight] = useState<number[]>([]);

  const toggleCurrentPlayer = () => {
    setCurrentPlayer((prevPlayer) => (prevPlayer === 'X' ? 'O' : 'X'));
  };

  const checkSquare = (index: number) => {
    if (data[index] !== '' || winner) return;
    const newData = [...data];
    newData[index] = currentPlayer;
    setData(newData);
    const result = getWinnerResult(newData);
    if (!result.winner) {
      toggleCurrentPlayer();
    }
    setWinner(result.winner);
    setCombHighLight(result.combHighLight);
  };

  const resetGame = () => {
    setData(DEFAULT_DATA);
    setWinner(null);
    setCombHighLight([]);
    setCurrentPlayer('X');
  };

  return (
    <GameContext.Provider
      value={{
        data,
        currentPlayer,
        winner,
        combHighLight,
        checkSquare,
        toggleCurrentPlayer,
        resetGame
      }}
    >
      {children}
    </GameContext.Provider>
  );
};

export default GameProvider;
