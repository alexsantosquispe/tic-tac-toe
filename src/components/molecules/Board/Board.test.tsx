import { render } from '@testing-library/react';
import GameProvider from '../../../context/GameProvider';
import { I18NWrapper } from '../../../tests/testsUtils';
import Board from './Board';

describe('Board', () => {
  it('should render the board', () => {
    const component = render(
      <I18NWrapper>
        <GameProvider>
          <Board />
        </GameProvider>
      </I18NWrapper>
    );

    expect(component).toMatchSnapshot();
  });
});
