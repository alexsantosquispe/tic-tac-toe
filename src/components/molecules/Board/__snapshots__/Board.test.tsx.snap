// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Board should render the board 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-[20rem] w-[20rem] rounded-2xl border border-neutral-200 bg-neutral-100 p-3 shadow-lg md:h-[28rem] md:w-[28rem]"
        data-testid="board"
      >
        <div
          aria-label="Tic Tac Toe board"
          class="flex flex-1 flex-wrap overflow-hidden rounded-xl border border-dashed border-neutral-200 bg-white"
        >
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-0"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-1"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-2"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-3"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-4"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-5"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-6"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-7"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Empty"
            class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
            data-testid="square-8"
            type="button"
          >
            <span
              class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
            >
              <svg
                class="size-20 md:size-30 text-neutral-200/60"
                data-testid="x-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="3"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-[20rem] w-[20rem] rounded-2xl border border-neutral-200 bg-neutral-100 p-3 shadow-lg md:h-[28rem] md:w-[28rem]"
      data-testid="board"
    >
      <div
        aria-label="Tic Tac Toe board"
        class="flex flex-1 flex-wrap overflow-hidden rounded-xl border border-dashed border-neutral-200 bg-white"
      >
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-0"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-1"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-2"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-3"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-4"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-5"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-6"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-7"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Empty"
          class="group relative flex h-1/3 w-1/3 items-center justify-center border border-dashed border-neutral-200 p-1 transition-colors duration-150 hover:cursor-pointer hover:bg-neutral-50"
          data-testid="square-8"
          type="button"
        >
          <span
            class="absolute inset-0 flex items-center justify-center opacity-0 transition-opacity duration-150 group-hover:opacity-100"
          >
            <svg
              class="size-20 md:size-30 text-neutral-200/60"
              data-testid="x-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
